simulations/vip-152/vip-152/simulation.ts

```diff
@@ -0,0 +1,136 @@
+      WBNB = new ethers.Contract(WBNB_ADDRESS, ERC20_ABI, provider);
+      proxyAdmin = new ethers.Contract(PROXY_ADMIN, PROXY_ADMIN_ABI, provider);
+    });
+
+    it("validate admin", async () => {
+      expect(await vBNB.admin()).to.be.equal(VBNBAdmin_ADDRESS);
+    });
+
+    it("validate PSR", async () => {
+      expect(await vBNBAdmin.protocolShareReserve()).to.be.equal(PSR);
+    });
+
+    it("config check", async () => {
+      expect(await psr.totalDistributions()).to.be.equal(4);
+    });
+
+    it("ownership check", async () => {
+      expect(await psr.pendingOwner()).to.be.equal("0x0000000000000000000000000000000000000000");
+      expect(await psr.owner()).to.be.equal(NORMAL_TIMELOCK);
+    });
+
+    it("configurations", async () => {
+      let config = await psr.distributionTargets(0);
+      expect(config.schema).to.be.equal(0);
+      expect(config.percentage).to.be.equal(50);
+      expect(config.destination).to.be.equal(RISK_FUND);
+
+      config = await psr.distributionTargets(1);
+      expect(config.schema).to.be.equal(0);
+      expect(config.percentage).to.be.equal(50);
+      expect(config.destination).to.be.equal(TREASURY);
+
+      config = await psr.distributionTargets(2);
+      expect(config.schema).to.be.equal(1);
+      expect(config.percentage).to.be.equal(40);
+      expect(config.destination).to.be.equal(RISK_FUND);
+
+      config = await psr.distributionTargets(3);
+      expect(config.schema).to.be.equal(1);
+      expect(config.percentage).to.be.equal(60);
+      expect(config.destination).to.be.equal(TREASURY);
+    });
+
+    it("reduce reserves", async () => {
+      expect(await WBNB.balanceOf(PSR)).to.be.equal(ethers.utils.parseEther("0"));
+      await vBNBAdmin.reduceReserves(ethers.utils.parseEther("1"));
+      expect(await WBNB.balanceOf(PSR)).to.be.equal(ethers.utils.parseEther("1"));
+      await psr.releaseFunds(CORE_POOL_COMPTROLLER, [WBNB_ADDRESS]);
+      expect(await WBNB.balanceOf(PSR)).to.be.equal(ethers.utils.parseEther("0"));
+      expect(await WBNB.balanceOf(RISK_FUND)).to.be.equal("1093600150889113267");
+      expect(await WBNB.balanceOf(TREASURY)).to.be.equal("1093600150889113267");
+      await vBNBAdmin.reduceReserves("100");
+      expect(await WBNB.balanceOf(PSR)).to.be.equal(100);
+      await psr.releaseFunds(CORE_POOL_COMPTROLLER, [WBNB_ADDRESS]);
+      expect(await WBNB.balanceOf(RISK_FUND)).to.be.equal("1093600150889113317");
+      expect(await WBNB.balanceOf(TREASURY)).to.be.equal("1093600150889113317");
+    });
+
+    it("validate proxy admin", async () => {
+      expect(await proxyAdmin.getProxyAdmin(VBNBAdmin_ADDRESS)).to.be.equal(PROXY_ADMIN);
+      expect(await proxyAdmin.getProxyAdmin(PSR)).to.be.equal(PROXY_ADMIN);
+      expect(await proxyAdmin.getProxyAdmin(RISK_FUND)).to.be.equal(PROXY_ADMIN);
+
+      expect(await proxyAdmin.owner()).to.be.equal(NORMAL_TIMELOCK);
+    });
+  });
```

Comment:
Given we are updating the RiskFund implementation, I would check, before and after the VIP execution, the returned values of the function `assetsReserves` for the following assets (we release funds to the RiskFund recently [here](https://bscscan.com/tx/0xc74cfce3030f039432015a4f9a9257c1e6fc1c3d5ff8a8b02b0bcbf14088bb99)). These values shouldn't change with the upgrade.

```
0x0782b6d8c4551b9760e74c0545a9bcd90bdc41e5 HAY
0xd17479997f34dd9156deef8f95a52d81d265be9c USDD
0x55d398326f99059ff775485246999027b3197955 USDT
0x52f24a5e03aee338da5fd9df68d2b6fae1178827 ankrBNB
0x965f527d9159dce6288a2219db51fc6eef120dd1 BSW
0x4b0f1812e5df2a09796481ff14017e6005508003 TWT
0x12bb890508c125661e03b09ec06e404bc9289040 RACA
0xfb5b838b6cfeedc2873ab27866079ac55363d37e FLOKI
0x1bdd3cf7f79cfb8edbb955f20ad99211551ba275 BNBx
0xc2e9d07f66a89c44062459a47a0d2dc038e4fb16 stkBNB
0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c WBNB
0x352cb5e19b12fc216548a2677bd0fce83bae434b BTT
0xce7de646e7208a4ef112cb6ed5038fa6cc6b12e3 TRX
0xaef0d72a118ce24fee3cd1d43d383897d05b4e99 WINK
```